numba/annotations/step.py

```diff
@@ -0,0 +1,143 @@
+"""
+A light and fast template engine.
```

Comment:
I don't think it's a good idea to rely on a template engine that's not maintained, not really used, and apparently not Python 3 compatible. I think we should just add a dependency to jinja2 or mako (whichever you prefer).